features/256-auth-gear-interaction-overview-in-next/api.md

```diff
@@ -0,0 +1,126 @@
+# auth gear interaction overview in next
+
+Base on the new product architecture decision (auth gear + cloud function and drop record gear), skygear next should bring up a new experience on how auth gear is used in cloud function and client SDK.
+
+## Goal
+
+* Keep auth gear simple, focus on auth related functionalities.
+* Keep client SDK simple, less interaction with auth gear directly.
+* More cloud functions, a developer should create corresponding cloud funtion to fulfill application logic.
+
+## Major changes
+
+* Remove admin API in client SDK.
+* Remove user profile support in auth gear.
+
+## Architecture overview
+
+![](https://i.imgur.com/GQbup7u.png)
+
+- Cloud function has its own `user` table and it should save user profile (gender, age, ..., anything other than auth infomation) here.
+- Auth gear's `user` table saves auth related information only.
+
+## Use cases
+
+### client SDK
+
+* should support to access current user's attributes after login/signup successfully.
+* should support change password
+* should support forgot password flow
+* should support verify user
+
+### cloud function
+
+* should support to indicate it is restricted to be invoked by authenticated user only.
+* should support to have a table with user_id as foreign key.
+* should support to access cloud funtion user's attribute in context.
+* should support to set a user's role, disabled state.
+* should support to query user.
+
+## Auth related information
+
+Auth gear would focus on auth related information only, they are
+
+```go
+AuthData        map[string]string
+SignUpAt        string
+LastSeenAt      string
+LastLoginAt     string
+Disabled        bool
+DisabledMessage string
+DisabledExpiry  string
+Roles           []string
+Verified        bool
+```
+
+## How to update user attributes in client SDK
+
+Since client SDK _**won't**_ have API support to update user attributes directly, developer should create their own cloud function for client SDK to update user attributes accordingly.
+
+![](https://i.imgur.com/Awrge8l.png)
+
+And followings APIs will be dropped in client SDK:
+
+- `adminDisableUser`
+- `adminenableUser`
+- `adminResetPassword`
+- `assignUserRole`
+- `fetchUserRole`
+- `revokeUserRole`
+- `setAdminRole`
+- `setDefaultRole`
+
+## How to update user attributes in clound function
+
+For restricted attributes, cloud function should use corresponding interface in auth gear to update.
```

Comment:
I think this list is not completed too? For example Enable Users, Reset Password, etc..